tests/test_nblist.py

```diff
@@ -19,47 +20,84 @@ def test_empty_neighborlist():
         custom_ops.Neighborlist_f32(0)
 
 
-def test_block_bounds():
-    np.random.seed(2020)
-    sizes = [128, 156, 298]
-    max_size = max(sizes)
-    nblist = custom_ops.Neighborlist_f64(max_size)
-    for N in sizes:
-        nblist.resize(N)
-        block_size = 32
+def reference_block_bounds(coords: NDArray, box: NDArray, block_size: int) -> Tuple[NDArray, NDArray]:
+    # Make a copy to avoid modify the coordinates that end up used later by the Neighborlist
+    coords = coords.copy()
+    N = coords.shape[0]
+    box_diag = np.diagonal(box)
+    num_blocks = (N + block_size - 1) // block_size
 
-        D = 3
+    ref_ctrs = []
+    ref_exts = []
+
+    for bidx in range(num_blocks):
+        start_idx = bidx * block_size
+        end_idx = min((bidx + 1) * block_size, N)
+        block_coords = coords[start_idx:end_idx]
+        min_coords = block_coords[0]
+        max_coords = block_coords[0]
+        for new_coords in block_coords[1:]:
+            center = 0.5 * (max_coords + min_coords)
+            new_coords -= box_diag * np.floor((new_coords - center) / box_diag + 0.5)
+            min_coords = np.minimum(min_coords, new_coords)
+            max_coords = np.maximum(max_coords, new_coords)
+
+        ref_ctrs.append((max_coords + min_coords) / 2)
+        ref_exts.append((max_coords - min_coords) / 2)
+
+    ref_ctrs = np.array(ref_ctrs)
+    ref_exts = np.array(ref_exts)
+    return ref_ctrs, ref_exts
+
+
+@pytest.mark.parametrize("precision,atol,rtol", [(np.float32, 1e-6, 1e-6), (np.float64, 1e-7, 1e-7)])
+@pytest.mark.parametrize("sort", [True, False])
+def test_block_bounds_dhfr(precision, atol, rtol, sort):
+    _, _, coords, box = setup_dhfr()
+
+    if precision == np.float32:
+        nblist = custom_ops.Neighborlist_f32(coords.shape[0])
+    else:
+        nblist = custom_ops.Neighborlist_f64(coords.shape[0])
 
-        coords = np.random.randn(N, D)
-        box_diag = np.random.rand(3) + 1
-        box = np.eye(3) * box_diag
-        num_blocks = (N + block_size - 1) // block_size
+    if sort:
+        perm = hilbert_sort(coords + np.argmin(coords), coords.shape[1])
+        coords = coords[perm]
+
+    block_size = 32
```

Comment:
nit/future: does this need to be kept in sync with the kernel implementation? Maybe eventually could expose `NEIGHBORLIST_BLOCK_SIZE` or similar from the `custom_ops` module, similar to `FIXED_EXPONENT`?

Reply:
Never mind, sorry. I thought that it was hardcoded but now see that `block_size` is an argument to the kernel.